aws-distro-opentelemetry-node-autoinstrumentation/src/aws-metric-attribute-generator.ts

```diff
@@ -406,6 +459,12 @@ export class AwsMetricAttributeGenerator implements MetricAttributeGenerator {
           span.attributes[AwsSpanProcessingUtil.GEN_AI_REQUEST_MODEL]
         );
       }
+
+      if (cloudFormationIdentifier === undefined) {
+        cloudFormationIdentifier = remoteResourceIdentifier;
+      }
+
+      attributes[AWS_ATTRIBUTE_KEYS.AWS_CLOUDFORMATION_PRIMARY_IDENTIFIER] = cloudFormationIdentifier;
```

Comment:
Should this line be nested below in:
- `if (remoteResourceType !== undefined && remoteResourceIdentifier !== undefined) {`

I thought we originally wanted these 3 scenarios:
1. Set all 3: `AWS_REMOTE_RESOURCE_TYPE`, `AWS_REMOTE_RESOURCE_IDENTIFIER`, and `AWS_CLOUDFORMATION_PRIMARY_IDENTIFIER`
2. Set 2:  `AWS_REMOTE_RESOURCE_TYPE`, `AWS_REMOTE_RESOURCE_IDENTIFIER`
3. Set `none`

With the current logic, a 4th scenario is possible:
- Set only: `AWS_CLOUDFORMATION_PRIMARY_IDENTIFIER`

Do we want the 4th scenario? I imagine that `AWS_CLOUDFORMATION_PRIMARY_IDENTIFIER` must be dependent on the 2 other attributes (today we do not set either of those 2 attributes if one of those are missing)